lenskit-core/src/main/java/org/grouplens/lenskit/data/history/History.java

```diff
@@ -75,6 +75,8 @@ private History() {}
      * @return An empty history for the user.
      */
     @Nonnull
+    @SafeVarargs
+    @SuppressWarnings("varargs") // method is safe MDE 2015-05-08
     public static <E extends Event> UserHistory<E> forUser(long id, E... events) {
         List<E> list = ImmutableList.copyOf(events);
```

Comment:
I'm not going to pretend to fully grock the varargs safety issue, but this looks like the type of code that I understand to be potentially unsafe. It probably depends on whether ImutableList inspects the type of events at runtime, since (as I understand it) events would consider itself a Event array not an E array.